.github/workflows/cicd.yml

```diff
@@ -0,0 +1,136 @@
+name: "CICD"
+
+on:
+  # Test run before merging
+  pull_request:
+    branches:
+      - main
+  # On merged
+  push:
+    branches:
+      - main
+  # On released
+  release:
+    types: [created]
+
+jobs:
+  build-docker:
+    permissions:
+      id-token: write
+      contents: write
+    runs-on: ubuntu-22.04
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    name: "Build and push images"
+    strategy:
+      fail-fast: false
+      matrix:
+        application: [neo-savant]
+        include:
+          - application: neo-savant
+            docker_domain: asia-docker.pkg.dev
+            registry_stg: asia-docker.pkg.dev/prj-d-devops-services-4dgwlsse/zilliqa-private
+            image_name: neo-savant
+            path: products/neo-savant
+    env:
+      DEFAULT_BRANCH: main
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          fetch-depth: 0
+
+      - id: "auth"
+        name: "Authenticate to Google Cloud"
+        uses: "google-github-actions/auth@v1"
+        with:
+          token_format: "access_token"
+          workload_identity_provider: "${{ secrets.GCP_PRD_GITHUB_WIF }}"
+          service_account: "${{ secrets.GCP_PRD_GITHUB_SA_GCS_CACHE }}"
+          create_credentials_file: true
+
+      - name: Configure bazel GCS cache
+        run: |
+          export BAZEL_REMOTE_CACHE_CREDENTIALS_JSON='${{ steps.auth.outputs.credentials_file_path }}'
+          export BAZEL_REMOTE_CACHE_ENDPOINT='${{ vars.BAZEL_REMOTE_CACHE_ENDPOINT }}'
+          python config/gcp_cs_cache.py
+        shell: bash
+
+      - name: Preparing merged branches
+        # Fetch the ref of the base branch, just the single commit.
```

Comment:
Surely this should no longer be necessary? You don't need to commit your container versions back anymore if you're using z app ... ?